File: Sisteme_Distribuite/examen SD/ceva/SD/kotlin/SparkExample/src/main/kotlin/com/sd/laborator/ClientRDD.kt
package com.sd.laborator

import org.apache.spark.SparkConf
import org.apache.spark.streaming.Durations
import org.apache.spark.streaming.api.java.JavaStreamingContext


fun prelucrare(date:List<String>)
{
    if(date.size!=0)
    {
        println("Prelucrez date!")
        date.forEach{
            var s = it.split("{")
            s.forEach{
                var i = it.split("\n")
                println(i[1] +"\t" + i[2])
                var c = i[2].split(":")
                var max = c[1].toInt()
                if(max>400)
                {
                    println("Compania cu " + i[1] + "indeplineste conditia profit > 400")
                }
            }
        }
        println("Am terminat de prelucrat datele!")
    }
}

fun main(args: Array<String>)
{
    println("Creez configurez Spark")
    val sparkConf = SparkConf().setMaster("local").setAppName("Examen")
    val sparkContext = JavaStreamingContext(sparkConf, Durations.seconds(1))

    val streamSocket = sparkContext.socketTextStream("localhost", 1234)
    println("Astept sa primesc date")
    streamSocket.foreachRDD{ rdd->prelucrare(rdd.collect()) }

    sparkContext.start()
    sparkContext.awaitTermination()

}

File: Sisteme_Distribuite/examen SD/ceva/subiecte/server_python_RDD/ceba/client_rdd.py
from pyspark import SparkContext, SparkConf, StorageLevel
from pyspark.streaming import StreamingContext
import os

pozitiv =["ana", "python", "mere"]
negativ =["date", "time"]

def is_pozitif(words):
    if(words in pozitiv):
        return True
    return False

def is_negatif(words):
    if (words in negativ):
        return True
    return False

def read_list():
    pass
def prelucrare(word):
    '''if(is_pozitif(word)):
        print( "\tpozitiv\n")
    elif(is_negatif(word)):
        print( "\tnegativ\n")
    else:
        print( "\tneutru\n")'''
    print("catel")

def main():
    os.environ["PYSPARK_PYTHON"] = '/usr/bin/python3'
    #spark_conf = SparkConf.setMaster(value="local").setAppName("News")
    spark_context = SparkContext(master="local[*]")
    print("Ma pregatesc sa primesc date!")
    spark_stream = StreamingContext(spark_context, 2)
    stream_data = spark_stream.socketTextStream("localhost",1234 )
    print("trimesc date")
    stream_data.foreachRDD(lambda rdd:print(rdd.collect()))

    print("am terminat de primit date!")

    spark_stream.start()
    spark_stream.awaitTermination()
    spark_stream.stop()


if __name__ == '__main__':
        main()